### PR TITLE
feat(platform/bitbucket): modify getJsonFile to use branchOrTag on Bitbucket

### PR DESCRIPTION
**Pull request by olegkrivtsov**
## Changes:

modify getJsonFile to use branchOrTag on Bitbucket

## Context:

Related to #2496 and PR https://github.com/renovatebot/renovate/pull/12514

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/platform/bitbucket/index.ts [hunk 1]
```diff
@@ -116,7 +116,8 @@ export async function getRawFile(
   // See: https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/src/%7Bcommit%7D/%7Bpath%7D
   const repo = repoName ?? config.repository;
   const path = fileName;
-  const url = `/2.0/repositories/${repo}/src/HEAD/${path}`;
+  const url =
+    `/2.0/repositories/${repo}/src/` + (branchOrTag || `HEAD`) + `/${path}`;
```
Comment (viceice) [inline lib/platform/bitbucket/index.ts:120]: I think this needs some real tests, as i'm not sure if branches needs to be encoded, or if we need to fetch the commit sha first

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/platform/bitbucket/index.ts (hunk 1 above)
Comment (olegkrivtsov) [reply lib/platform/bitbucket/index.ts:120]: Hi @viceice I've done real testing on my repo https://bitbucket.org/olegkrivtsov/renovate-test. 

I used these API calls for getting the file contents:

curl --request GET --url 'https://olegkrivtsov:XXXXXXXX@api.bitbucket.org/2.0/repositories/olegkrivtsov/renovate-test/src/dev/README.md'

curl --request GET --url 'https://olegkrivtsov:m3DgrS7DYRBKgMVQX9yG@api.bitbucket.org/2.0/repositories/olegkrivtsov/renovate-test/src/00fa6d4b0ebcb3bcd9940a0cfe436185a2e32b66/README.md'

Both of them worked, so it seems we can pass both branch name (no encoding is needed), and SHA1 of a commit.

Comment (viceice) [reply lib/platform/bitbucket/index.ts:120]: Can you test a branch with slashes?

Comment (olegkrivtsov) [reply lib/platform/bitbucket/index.ts:120]: Hi @viceice I've tested a branch with slash `feat/123-test`:

```
curl --request GET --url 'https://olegkrivtsov:XXXXX@api.bitbucket.org/2.0/repositories/olegkrivtsov/renovate-test/src/feat/123-test/README.md'
```

This didn't work.

I tried to url-encode the branch name, and it didn't work as well:

```
curl --request GET --url 'https://olegkrivtsov:gaM5YcN8nw9fyd7Esvvd@api.bitbucket.org/2.0/repositories/olegkrivtsov/renovate-test/src/feat%2F123-test/README.md'
```

I guess that maybe getting the SHA1 of a HEAD commit of the branch and passing it to the URL would work?

Comment (viceice) [reply lib/platform/bitbucket/index.ts:120]: ok, so branches whithout slash can used directly and others needs to fetch head commit first. Can you implement this?

**Review by viceice — COMMENTED** [review]
see other comment

File: lib/platform/bitbucket/index.ts (hunk 1 above)
Comment (olegkrivtsov) [reply lib/platform/bitbucket/index.ts:120]: @viceice sure, I'll implement this.

File: lib/platform/bitbucket/index.ts [hunk 2]
```diff
@@ -116,7 +118,17 @@ export async function getRawFile(
   // See: https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/src/%7Bcommit%7D/%7Bpath%7D
   const repo = repoName ?? config.repository;
   const path = fileName;
-  const url = `/2.0/repositories/${repo}/src/HEAD/${path}`;
+
+  let finalBranchOrTag = branchOrTag;
+  if (branchOrTag && branchOrTag.indexOf(pathSeparator) !== -1) {
```
Comment (viceice) [inline lib/platform/bitbucket/index.ts:123]:
```suggestion
  if (branchOrTag?.includes(pathSeparator)) {
```
A little bit simpler  condition. 😉

**Review by viceice — APPROVED** [review]

**Pull request merged**